File: Backend/htbp_parent/service/service_hosp/src/main/java/com/seclib/htbp/hosp/controller/HospitalController.java
package com.seclib.htbp.hosp.controller;

import com.seclib.htbp.common.result.Result;
import com.seclib.htbp.hosp.service.HospitalService;
import com.seclib.htbp.model.hosp.Hospital;
import com.seclib.htbp.vo.hosp.HospitalQueryVo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/admin/hosp/hospital")
@CrossOrigin
public class HospitalController {

    @Autowired
    private HospitalService hospitalService;

    @GetMapping("list/{page}/{limit}")
    public Result listHosp(@PathVariable Integer page, @PathVariable Integer limit, HospitalQueryVo hospitalQueryVo) {
        Page<Hospital> pageModel = hospitalService.selectHospPage(page,limit,hospitalQueryVo);

        return Result.ok(pageModel);
    }

    @GetMapping("updateHospStatus/{id}/{status}")
    public  Result updateHospStatus(@PathVariable String id,@PathVariable Integer status){
        hospitalService.updateStatus(id,status);
        return Result.ok();
    }

    @GetMapping("showHospitalDetail/{id}")
    public Result showHospitalDetail(@PathVariable String id){
        Map<String,Object> result = hospitalService.getHospById(id);
        return Result.ok(result);
    }

}

File: Backend/htbp_parent/service/service_hosp/src/main/java/com/seclib/htbp/hosp/controller/HospitalController.kt
package com.seclib.htbp.hosp.controller

import com.seclib.htbp.common.result.Result
import com.seclib.htbp.hosp.service.HospitalService
import com.seclib.htbp.vo.hosp.HospitalQueryVo
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController


@RestController
@RequestMapping("/admin/hosp/hospital") //@CrossOrigin
class HospitalController {
    @Autowired
    private val hospitalService: HospitalService? = null
    @GetMapping("list/{page}/{limit}")
    fun listHosp(@PathVariable page: Int?, @PathVariable limit: Int?, hospitalQueryVo: HospitalQueryVo?): Result<*> {
        val pageModel = hospitalService!!.selectHospPage(page, limit, hospitalQueryVo)
        return Result.ok(pageModel)
    }

    @GetMapping("updateHospStatus/{id}/{status}")
    fun updateHospStatus(@PathVariable id: String?, @PathVariable status: Int?): Result<*> {
        hospitalService!!.updateStatus(id, status)
        return Result.ok<Any>()
    }

    @GetMapping("showHospitalDetail/{id}")
    fun showHospitalDetail(@PathVariable id: String?): Result<*> {
        val result = hospitalService!!.getHospById(id)
        return Result.ok(result)
    }
}